### ISSUE TITLE
Generic type is array

### ISSUE THREAD

**Issue by mhagnumdw** [typestack/class-transformer#395]
**I was trying to...**
 
 

I have the generic class below.

```ts
export class OpenShiftData<T> {

  @Type(options => {
    return (options.newObject as OpenShiftData<T>).type;
  })
  data: T;

  messages: Dictionary;

  get messagesString(): string | null {
    if (!this.messages) {
      return null;
    }
    return Object.keys(this.messages)
      .map(msgKey => `${msgKey}: ${this.messages[msgKey]}`)
      .join('; ');
  }

  @Exclude()
  private type: Function;

  constructor(type: Function) {
    this.type = type;
  }

}
```

And this class

```ts
export class Pod {

  name: string;

  status: string;

  ready: boolean;

  // I want to use this method !!!
  get validStatus(): string {
    if (this.status === 'Running' && this.ready === false) {
      return 'RunningNotReady';
    }
    return this.status;
  }

}
```

**The problem:**

But my generic type `T` is an array of type `Pod[]`, like this: `OpenShiftData<Pod[]>`

Is it possible to make this transformation?

**Comment by NoNameProvided** [comment]
This is a question about Typescript typing usage, not the library, also you didn't post the part where you actually trying to use it. Anyway, please ask questions like this on Stackoverflow. 

PS: First please use the template to report issues, ask questions, etc.

**Comment by mhagnumdw** [comment]
Hi!

Sorry.

I opened the issue here because there is an option "Question". I used the template.  Option "New Issue" > "Question" > "Get started"

I forgot to put the code where I tried to use it (sorry again). Here (for someone who will look into this issue in the future):

In theory the usage should be something like:

```ts
plainToClassFromExist(new OpenShiftData<Pod[]>(Pod[]), result))
```

But I obviously can't pass the `Pod[]` type. The `Pod` type is accepted, but it doesn't work for me.

Tks!
